File: src/main/java/com/letgo/book/application/find/FindBookQueryHandler.java
package com.letgo.book.application.find;

import com.letgo.book.domain.Book;
import com.letgo.book.domain.BookFinder;
import com.letgo.book.domain.BookId;
import com.letgo.shared.application.bus.query.QueryHandler;
import org.springframework.stereotype.Service;

@Service
public class FindBookQueryHandler implements QueryHandler<FindBookQuery> {
    private final BookFinder finder;

    public FindBookQueryHandler(BookFinder finder) {
        this.finder = finder;
    }

    public FindBookQueryResponse handle(FindBookQuery query) {
        Book book = finder.find(BookId.create(query.id()));
        return new FindBookQueryResponse(book.title());
    }
}

File: src/main/java/com/letgo/book/application/find/FindBookQueryHandler.kt
package com.letgo.book.application.find

import com.letgo.book.application.BookResponse
import com.letgo.book.application.BookResponseMapper
import com.letgo.book.domain.BookFinder
import com.letgo.book.domain.BookId
import com.letgo.shared.application.bus.query.QueryHandler

class FindBookQueryHandler(
    private val finder: BookFinder,
    private val mapper: BookResponseMapper,
) : QueryHandler<FindBookQuery, BookResponse> {
    override fun handle(query: FindBookQuery): BookResponse {
        return finder.find(BookId(query.id)).let { mapper.map(it) }
    }
}
